zerver/webhooks/front/view.py

```diff
@@ -0,0 +1,188 @@
+from typing import Any, Dict, Optional, Text, Tuple
+
+from django.http import HttpRequest, HttpResponse
+from django.utils.translation import ugettext as _
+
+from zerver.decorator import api_key_only_webhook_view
+from zerver.lib.actions import check_send_stream_message
+from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.response import json_error, json_success
+from zerver.models import UserProfile
+
+def get_message_data(payload: Dict[Text, Any]) -> Optional[Tuple[Text, Text, Text, Text]]:
+    try:
+        link = "https://app.frontapp.com/open/" + payload['target']['data']['id']
+        outbox = payload['conversation']['recipient']['handle']
+        inbox = payload['source']['data'][0]['address']
+        subject = payload['conversation']['subject']
+    except KeyError:
+        return None
+
```

Comment:
Okay, so handling a `KeyError` like this is bad. We do not want an error to go unnoticed. There are a couple of reasons for this:
1. We have a logger that logs all `KeyErrors` in a file so that we can fix them. So if something fails, we ought to know about it and it shouldn't just fail silently unless it is super unimportant.
2. Even if we choose to handle a `KeyError` in this fashion, it makes sense to minimise the code that goes inside the `try` block as much as possible. One silent error is better than four, for instance!
3. Also, with webhook payloads, when you have a function like this, you want to be sure that it will have the values that you are looking for, so this kind of error checking should not be needed at all. And if you look at some of our more complex integrations (such as Gitlab), we rarely respond to `KeyErrors` in this manner.

For now, I would recommend not doing any error checking at all. If you know for sure that a certain payload might not have certain values, you should account for that in other ways (for example, checking to see if the payload contains them and then having a message that is different in the scenario that it doesn't). We can have a longer conversation about this on `chat.zulip.org` if you'd like! :)